### PR TITLE
Feature gff to gtf

### PR DESCRIPTION
**Pull request by JenkoB**
Add gff3 to gtf convertor.

### CODE REVIEW

**Comment by mstajdohar** [comment]
Fix tests.

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-docstring,invalid-name,redefined-outer-name
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: Remove this 2 lines and fix lint errors.

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+def main():
+    args = parse_arguments()
+
+    inFile = open(args.gff3_file, 'r')
+
+    with open(args.gtf_file, 'w') as f:
+        for line in inFile:
+            # skip comment lines that start with the '#' character
+            if line[0] != '#':
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: Call `.strip()` before this check:
```
line = line.strip()
```

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                # split line into columns by tab
+                data = line.strip().split('\t')
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: No need to call `.strip()` here if you call it above.

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+                ID = ''
+                # if the feature is a gene
+                if data[2] == "gene":
+                    # get the id
+                    ID = data[-1].split('ID=')[-1].split(';')[0]
+
+                    # if the feature is anything else
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: Move this comment to the line below:
```
else:  # feature is anything but gene
```

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    with open(args.gtf_file, 'w') as f:
+        for line in inFile:
+            # skip comment lines that start with the '#' character
+            if line[0] != '#':
+                # split line into columns by tab
+                data = line.strip().split('\t')
+
+                ID = ''
+                # if the feature is a gene
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: Move this if to the line below
```
if data[2] == "gene":  # feature is gene
```

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                if data[2] == "gene":
+                    # get the id
+                    ID = data[-1].split('ID=')[-1].split(';')[0]
+
+                    # if the feature is anything else
+                else:
+                    # get the parent as the ID
+                    ID = data[-1].split('Parent=')[-1].split(';')[0]
+
+                    # modify the last column
```
Comment (mstajdohar) [inline resolwe_bio/tools/gff3_to_gtf.py]: Align the comment with line below.

File: resolwe_bio/processes/support_processors/gff_to_gtf.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    - name: gtf
+      label: Converted gtf file
+      type: basic:file
+    - name: source
+      label: Gene ID source
+      type: basic:string
+  run:
+    runtime: polyglot
+    bash: |
+
```
Comment (mstajdohar) [inline resolwe_bio/processes/support_processors/gff_to_gtf.yml]: Unnecessary empty line.

**Comment by mstajdohar** [comment]
Remove the `Add Subread process` commit and rebase to master. Use the interactive rebase (`git rebase -i HEAD~3`).

**Comment by mstajdohar** [comment]
@jkokosar, please review.

File: resolwe_bio/processes/support_processors/gff_to_gtf.yml [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# ============
+# Gff to gtf
+# ============
+
+---
+
+- slug: gff-to-gtf
+  name: Annotation (GFF3)
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/gff_to_gtf.yml]: Name of the processor is not descriptive enough. Use something like "Convert GFF to GTF"

File: resolwe_bio/processes/support_processors/gff_to_gtf.yml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  data_name: 'gff'
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/gff_to_gtf.yml]: Include input file name in the data_name. Maybe use `"Convert to GFF ('{{ src.file }}')"`

File: resolwe_bio/processes/support_processors/gff_to_gtf.yml [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  version: 0.0.1
+  type: data:annotation
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/gff_to_gtf.yml]: Set type to `data:annotation:gtf`

File: resolwe_bio/tools/gff3_to_gtf.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+"""Convert ggf3 to gtf files."""
```
Comment (jkokosar) [inline resolwe_bio/tools/gff3_to_gtf.py]: fix typo

**Pull request merged**